### PR TITLE
Fix index worker depletion

### PR DESCRIPTION
**Pull request by tobim**
###  :notebook_with_decorative_cover: Description

A bug that causes the removal of index workers in case of false positive candidate partitions is fixed here.

###  :memo: Checklist

- [x] All user-facing changes have changelog entries.

### CODE REVIEW

**Review by dominiklohmann — APPROVED** [review]
This needs a changelog entry, but other than that you're good to go. I actually noticed this while working on #1211 as well—and please let this be the last PR we merge before that one.

(Of if you want to do me a favor, wait until we merged #1211)

**Comment by tobim** [comment]
@dominiklohmann I added a second change that I thought makes sense. Please have another look.

File: libvast/src/system/index.cpp
Comment (dominiklohmann) [inline libvast/src/system/index.cpp:700]: I think we should capture `worker = std::move(*worker)` in the lambda that we pass to `await_evaluation_maps`. This code does not need to know about the `optional` wrapping.

File: libvast/src/system/index.cpp
Comment (dominiklohmann) [inline libvast/src/system/index.cpp:688]: This is great, but please move it back to the very top of the function.

**Review by dominiklohmann — APPROVED** [review]
I really like this additional change, just two nits. I did not verify this to work locally.

File: libvast/src/system/index.cpp
Comment (dominiklohmann) [reply libvast/src/system/index.cpp:688]: Actually, now that I think about it: Don't do that.

**Pull request merged**
